### PR TITLE
ovmf: various improvements (IPv6, TLS, debugging)

### PR DESCRIPTION
**Pull request by RaitoBezarius**
###### Description of changes

This brings various OVMF (and EDK2) improvements:

- (PXE) IPv6 enabled by default for OVMF.
- TLS support if needed
- Debug with serial port / source debugging

I wished I knew a way to wire `pkgs.enableDebugging` stuff right into that, but I am afraid I don't.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by nikstur — APPROVED** [review]
~~LGTM~~

File: pkgs/applications/virtualization/OVMF/default.nix [hunk 1]
```diff
@@ -3,6 +3,9 @@
 , secureBoot ? false
 , httpSupport ? false
 , tpmSupport ? false
+, tlsSupport ? false
+, debug ? false
+, sourceDebug ? debug
```
Comment (nikstur) [inline pkgs/applications/virtualization/OVMF/default.nix:8]:
```suggestion
, sourceDebug ? false
```

**Review by nikstur — CHANGES_REQUESTED** [review]

File: pkgs/applications/virtualization/OVMF/default.nix (hunk 1 above)
Comment (RaitoBezarius) [reply pkgs/applications/virtualization/OVMF/default.nix:8]: Usually, you'd expect that `debug` enables also source-level debugging in the binary, any reason to commit your suggestion?

Comment (nikstur) [reply pkgs/applications/virtualization/OVMF/default.nix:8]: Sorry my brain was completely fried apparently.  In my head this was the literal string "debug". What you did makes perfect sense.

**Review by nikstur — APPROVED** [review]

File: pkgs/applications/virtualization/OVMF/default.nix (hunk 1 above)
Comment (RaitoBezarius) [reply pkgs/applications/virtualization/OVMF/default.nix:8]: No worries!

**Pull request merged**